src/engine/controls/cuecontrol.cpp

```diff
@@ -1979,10 +1989,12 @@ double CueControl::quantizeCuePoint(double cuePos) {
         return cuePos;
     }
 
-    double closestBeat = pBeats->findClosestBeat(cuePos);
+    const auto cuePosition = mixxx::audio::FramePos::fromEngineSamplePos(cuePos);
+    const auto closestBeatPosition = pBeats->findClosestBeat(cuePosition);
     // The closest beat can be an unreachable  interpolated beat past the end of
     // the track.
-    if (closestBeat != -1.0 && closestBeat <= total) {
+    const double closestBeat = closestBeatPosition.toEngineSamplePos();
```

Comment:
Is it safe to invoke `toEngineSamplePos()` on `kInvalidValue`? Do we have a test for this use case that verifies the expected outcome?

I am aware that the value is not used if invalid. But since we invoke this function beforehand we should verify this.

Reply:
Not yet, this is taken care of in #4061. Also see the comment below.